source/mir/math/stat.d

```diff
@@ -461,6 +443,353 @@ unittest
     assert(float.max.repeat(3).hmean!(double, "fast").approxEqual(float.max));
 }
 
+/++
+Output range for gmean.
++/
+struct GMeanAccumulator(T, ProdAlgo prodAlgo) 
+    if (isMutable!T)
+{
+    import mir.math.numeric: ProdAccumulator;
+
+    ///
+    size_t count;
+    ///
+    ProdAccumulator!(T, prodAlgo) prodAccumulator;
+
+    ///
+    F gmean(F = T)() @property
+    {
+        return (cast(F) prodAccumulator.prod) ^^ (cast(F) 1 / cast(F) count);
```

Comment:
Please replace `^^` with `mir.math.common: pow` everywhere in the PR. `^^` is slow and depends on `std.math`.